### DISCUSSION TITLE
[How to] Create Uno Platform JetBrains Rider File Templates

### DISCUSSION THREAD

**Discussion by RedEye-Developers** [unoplatform/uno#18976] [Show and tell]
Sample!

![Screenshot (44)](https://github.com/user-attachments/assets/60c4947b-136a-4389-82f0-a91a128f12e3)

!!!Search in JetBrains Rider Settings "File Template"!!!

Page.xaml
![Screenshot (33)](https://github.com/user-attachments/assets/684b2760-61e8-4f69-a183-bdd45768a2ac)
Same Procedure All!
![Screenshot (36)](https://github.com/user-attachments/assets/e49509ee-d0be-43e7-9777-0370ca747116)
![Screenshot (37)](https://github.com/user-attachments/assets/ab76d001-0844-46b2-8c63-78ff3f4aaa34)
```xml
<Page
    x:Class="$NAMESPACE$.$CLASS$"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:$NAMESPACE$"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
       <!-- Make sure to create a $CLASS$.xaml.cs file using the "page" snippet -->
       
    </Grid>
</Page>
```

ResourceDictionary.xaml
![Screenshot (34)](https://github.com/user-attachments/assets/57c51ac3-e527-45d1-b790-95c9b6cfc2b2)
```xml
<ResourceDictionary
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    
</ResourceDictionary>
```

UserControl.xaml
![Screenshot (35)](https://github.com/user-attachments/assets/caac39e2-a41f-41d9-8a06-175cb7541a64)
![Screenshot (38)](https://github.com/user-attachments/assets/f2abed0e-7eeb-4d3b-aaea-7db50d80f0b3)
![Screenshot (39)](https://github.com/user-attachments/assets/54600b6f-de90-4d3b-ab6c-c8c07f90e864)
```xml
<UserControl
    x:Class="$NAMESPACE$.$CLASS$"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:$NAMESPACE$"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    d:DesignHeight="300"
    d:DesignWidth="400">

    <Grid>
       <!-- Make sure to create a $CLASS$.xaml.cs file using the "usercontrol" snippet -->
       
    </Grid>
</UserControl>
```

Page.cs
![Screenshot (40)](https://github.com/user-attachments/assets/021c4111-58f8-4403-a5a7-6bd7b2dc3936)
![Screenshot (42)](https://github.com/user-attachments/assets/b49a07f6-a519-45bb-87e9-1135866dc2f9)
```cs
namespace $NAMESPACE$;

public sealed partial class ClassName : Page
{
    public ClassName()
    {
        this.InitializeComponent();     
    }
}
```

UserControl.cs
![Screenshot (41)](https://github.com/user-attachments/assets/6c639750-f09a-428f-a773-f86995d52817)
![Screenshot (43)](https://github.com/user-attachments/assets/5bdd2ad7-f7c4-4bc7-ac84-5078bd6eeffe)

```cs
namespace $NAMESPACE$;

public sealed partial class ClassName : UserControl
{
    public ClassName()
    {
        this.InitializeComponent();
    }
}
```

MUST READ!

for Page.cs and UserControl.cs
Class Name and Constrictor Name You need to replace manually.

**Comment by jeromelaban** [comment]
Thanks for sharing your technique!
